src/main/java/com/vadmack/mongodbtest/service/ProjectService.java

```diff
@@ -23,24 +25,27 @@
     private final ProjectRepository repository;
     private final SequenceGeneratorService sequenceGeneratorService;
     private final PageableService pageableService;
+    private final SortService sortService;
 
     private final ModelMapper modelMapper = new ModelMapper();
 
     public List<ProjectDto> findList(
             @Nullable String namePart,
             @Nullable Integer pageNumber,
-            @Nullable Integer pageSize) {
-        String regex;
-        if (namePart == null) {
-            regex = "*";
-        } else {
-            regex = "*" + namePart + "*";
-        }
+            @Nullable Integer pageSize,
+            @Nullable String[] sortBy
+    ) {
+        String regex = namePartToRegex(namePart);
         pageableService.validateParams(pageNumber, pageSize);
         if (pageNumber != null && pageSize != null) {
+            if (sortBy == null){
+                sortBy = new String[1];
+                sortBy[0] = "id:0";
+            }
+            Sort sort = sortService.createSort(sortBy);
             return repository.findAllByNameLikeIgnoreCase(
                     regex,
-                    PageRequest.of(pageNumber, pageSize))
+                    PageRequest.of(pageNumber, pageSize, sort))
```

Comment:
Why you ignore "sort" in case if "pageNumber" and "pageSize" are not provided

Reply:
I suggest to build request to database using "MongoTemplate". It would be more flexible an "beautiful" solution in this case